src/views/Home.vue

```diff
@@ -2,59 +2,121 @@
+          :lat="position.latitude"
+          :lng="position.longitude"
+        >
+        </user-location>
       </div>
     </v-row>
+    <v-btn top right absolute @click="show = !show" style="z-index: 3" icon>
+      <v-icon x-large color="dark">mdi-bus-marker</v-icon>
+    </v-btn>
+    <buslines-card :show="show" />
   </v-container>
 </template>
 
 <script>
 import BusStopInfo from "../components/BusStopInfo";
+import MapMarker from "../components/MapMarker";
+import UserLocation from "../components/UserLocation";
+import BuslinesCard from "../components/BuslinesCard";
+import { mapGetters } from "vuex";
+
 export default {
   name: "Home",
-  components: { BusStopInfo },
+  components: { BuslinesCard, UserLocation, MapMarker, BusStopInfo },
+
   data() {
     return {
       selected: [],
-      buslines: [
-        { number: 1, check: false },
-        { number: 2, check: false },
-        { number: 3, check: false },
-        { number: 4, check: false },
-        { number: 5, check: false },
-        { number: 6, check: false },
-        { number: 7, check: false },
-        { number: 8, check: false },
-        { number: 9, check: false },
-        { number: 20, check: false },
-        { number: 24, check: false },
-        { number: 55, check: false },
-        { number: 34, check: false }
-      ]
+      map: null,
+      dialog: false,
+      stationName: null,
+      stationId: null,
+      show: false,
+      markerCluster: null,
+
+      position: { longitude: NaN, latitude: NaN }
     };
+  },
+  computed: {
+    ...mapGetters(["stations", "currentStation"])
+  },
+  watch: {
+    currentStation() {
+      this.openBusStop(this.currentStation.name, this.currentStation.id);
+    }
+  },
+  mounted() {
+    this.map = new window.google.maps.Map(this.$refs["map"], {
+      center: { lat: 51.202709, lng: 16.15371 },
+      zoom: 14,
+      disableDefaultUI: true,
+      styles: [
+        {
+          featureType: "poi",
+          elementType: "labels",
+          stylers: [{ visibility: "off" }]
+        }
+      ]
+    });
+    this.$mapService.getBuslines().then(() => {});
+    if (navigator.geolocation) {
+      navigator.geolocation.getCurrentPosition(
+        this.getPositionSuccess,
+        this.getPositionError
+      );
+    }
+  },
+
+  methods: {
+    changeBusStopProps(stationName, stationId) {
+      this.stationName = stationName;
+      this.stationId = stationId;
+    },
+    openBusStop(stationName, stationId) {
+      this.changeBusStopProps(stationName, stationId);
+      this.dialog = !this.dialog;
+    },
+    getPositionSuccess(pos) {
+      this.position = pos.coords;
+    },
+    getPositionError(err) {
+      console.warn(`ERROR(${err.code}): ${err.message}`);
+    },
+    getMap(callback) {
+      let vm = this;
+
+      function checkForMap() {
+        if (vm.map) callback(vm.map);
+        else setTimeout(checkForMap, 200);
+      }
+
+      checkForMap();
+    }
```

Comment:
explain it, please

Reply:
I need it to minimalize error which occurs from Google Map Api. Sometimes map doesn't load as quick as I want to. When i want for example load in mount instance markers and the map hasn't loaded yet it wait and check map status again. You have and example below.

```
this.$parent.getMap(map => {
      this.marker = new window.google.maps.Marker({
        position: { lat: this.lat, lng: this.lng },
        map: map,
        icon: image
      });
```

Reply:
We came to an agreement in private messages, replacing nested functions by async await